internal/gengapic/gengapic.go

```diff
@@ -531,3 +590,38 @@ func parseRequestHeaders(m *descriptor.MethodDescriptorProto) ([][]string, error
 
 	return matches, nil
 }
+
+// Determine whether routing annotation exists
+func dynamicRequestHeadersExist(m *descriptor.MethodDescriptorProto) bool {
+	dynamicHeaders := proto.HasExtension(m.GetOptions(), annotations.E_Routing)
+	if m == nil || m.GetOptions() == nil {
+		return false
+	}
+	return dynamicHeaders
```

Comment:
`dynamicHeaders` should be a `bool` (is `HasExtension` rather than `GetExtension`). 
Good call - I added a test for `dynamicRequestHeadersExist` (returns `true` if extension is set but empty, which is intended).